### PR TITLE
fix header_format_default access code search method

### PR DESCRIPTION
**Pull request by lawchange**
1.only search bits from input
2.reset d_data_reg and d_data_reg_len if access_code matched
3.fix overflow?

### CODE REVIEW

**Comment by mormj** [comment]
@lawchange - thank you for the contribution.  This looks very similar to #3692 - can you verify whether that pull request achieves the same goal?

File: gr-digital/include/gnuradio/digital/header_format_default.h [hunk 1]
```diff
@@ -178,6 +178,7 @@ class DIGITAL_API header_format_default : public header_format_base
     uint16_t d_bps; //!< bits/sec of payload modulation
 
     unsigned long long d_data_reg; //!< used to look for access_code
+    size_t d_data_reg_len; //!< Length of the saved access_code
```
Comment (willcode) [inline gr-digital/include/gnuradio/digital/header_format_default.h:181]: //!< Number of valid bits in data register

File: gr-digital/lib/header_format_default.cc [hunk 2]
```diff
@@ -112,23 +113,27 @@ bool header_format_default::parse(int nbits_in,
             while (nbits_processed < nbits_in) {
                 // shift in new data
                 d_data_reg = (d_data_reg << 1) | ((input[nbits_processed++]) & 0x1);
+                if (d_data_reg_len < d_access_code_len){
+                    ++d_data_reg_len;
+                }
+                if (d_data_reg_len == d_access_code_len){
```
Comment (willcode) [inline gr-digital/lib/header_format_default.cc:119]: Use else?

**Review by willcode — CHANGES_REQUESTED** [review]
Looks good after a couple of very minor changes.

**Comment by willcode** [comment]
@mormj This is a different piece of code than #3692 isn't it?

The check failed on a pybind header mismatch.

**Comment by mormj** [comment]
Yes, you're right it is - made too cursory a glance.  To fix the pybind headers, since nothing was changed in the API, you can just update the hashes in the _python.cc

```
md5sum gr-digital/include/gnuradio/digital/header_format_default.h
```
replace the hash in gr-digital/python/bindings/header_format_default_python.cc

**Comment by mbr0wn** [comment]
Due to lack of activity, I'll close this. Please reopen if that was an error on my part, we're certainly very happy about all contributions. If you do, please make sure formatting and DCO (sign-off) are in order. Thanks!

**Pull request closed without merging**